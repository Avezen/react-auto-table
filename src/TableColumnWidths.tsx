import React from 'react';

interface TableColumnWidthsProps {
    colWidths?: number[];
}

export const TableColumnWidths = ({
                          colWidths,
                      }: TableColumnWidthsProps) => (
    <React.Fragment>
        {colWidths && colWidths.length > 0 && (
            <colgroup>
                {colWidths.map((item, key) => (
                    <col key={key} width={item}/>
                ))}
            </colgroup>
        )}
    </React.Fragment>
);
