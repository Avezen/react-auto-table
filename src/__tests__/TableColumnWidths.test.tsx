import React from "react";
import {TableColumnWidths, TableColumnWidthsProps} from "../TableColumnWidths";
import renderer from 'react-test-renderer';

function renderTableColumnWidths(props: Partial<TableColumnWidthsProps> = {}) {
    const defaultProps: TableColumnWidthsProps = {
        colWidths: [12, 23]
    };

    return <TableColumnWidths {...defaultProps} {...props} />;
}

describe("<TableColumnWidths />", () => {
    test("should render component", async () => {
        const findByTestId = renderTableColumnWidths(),
            TableColumnWidthsComponent = renderer.create(findByTestId).toJSON();

        expect(TableColumnWidthsComponent).toMatchSnapshot();
    });
});
