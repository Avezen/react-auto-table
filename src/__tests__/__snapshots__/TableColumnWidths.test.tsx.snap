// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableColumnWidths /> should render component 1`] = `
<colgroup>
  <col
    width={12}
  />
  <col
    width={23}
  />
</colgroup>
`;
